python/ribasim/ribasim/model.py

```diff
@@ -148,6 +149,47 @@ def _write_tables(self, directory: FilePath) -> None:
                 input_entry.write(directory, self.modelname)
         return
 
+    def validate_model(self):
+        """
+        Checks:
+        - Whether all node types in the node field are valid
+        - Whether the node IDs of the node_type fields are valid
+        """
```

Comment:
Should there also be a check for whether node IDs from model fields correspond to node types from node object? Maybe we should go to a single source of truth here.

Reply:
Yes, that would be a good addition.
Up to you if you want to create an issue for that or add it within this PR